vernac/classes.ml

```diff
@@ -516,7 +513,7 @@ let do_instance_program env env' sigma ?hook ~global ~poly cty k u ctx ctx' pri
       Some term, termtype, sigma in
   let termtype, sigma = do_instance_resolve_TC term termtype sigma env in
   if not (Evd.has_undefined sigma) && not (Option.is_empty opt_props) then
-    let term = to_constr sigma (Option.get term) in
+    let term = Option.get term in
```

Comment:
While you're at it you could remove the option wrapping here

Reply:
Indeed, that was a good idea, done.